docs/proposals/rescheduling/README.md

```diff
@@ -0,0 +1,220 @@
+
+The cluster administrators want to ensure that the application can run normally, regardless of which cluster the application is running. When the application is unavailable,
+they want Karmada to reschedule the application and avoid clusters that failed to run before.
+
+## Goals
+
+- Extend the API of PropagationPolicy to provide users with the rescheduled configuration.
+- Extend the API of ResourceBinding to record previously evicted clusters.
+- Propose the implementation ideas for involved components, including `karmada-controller-manager`, `karmada-webhook` and `karmada-scheduler`.
+For example, `karmada-controller-manager` need to add a component to observe the health status of the application. `Karmada-scheduler` need to
+add a plugin to filter the evicted cluster.
+
+## Non-Goals
+
+## Proposal
+
+### User Stories (Optional)
+
+#### As a user, I use the cloud vendor's spot instance to deploy my application. As a result, the application may fail due to insufficient resources.
+
+Spot instances are a mode of instances that feature a discounted price and a system interruption mechanism, which means that you can purchase a spot instance ar a discounted price, but the system may automatically repossess it.
+When users use spot instances to deploy applictions, the probability of an application failing in a vendor's cluster increases significantly. This can also occur in clusters with resource preemption and reclamation.
+At this time, users hope that Karmada provides a rescheduling mechanism.
+
+### As a user, I deploy an application which depends on cluster infrastructure(such as network or storage).
+
+Supposing I deploy an application which depends on cluster infrastructure. Due to the sudden failure of the storage of member1, the application located in member1 suddenly fails.
+However, member1 is still available in Karmada'view and other applications which do not depend on storage work well.
+At this time, I want to be able to failover this single application.
+
+### Risks and Mitigations
+
+This proposal maintains the backward compatibility, the system built with previous versions of Karmada can be seamlessly migrated to the new version.
+The previous configurations(yamls) could be applied to the new version of Karmada and without any behavior change.
+
+## Design Details
+
+### API change
+
+Add related rescheduling configuration:
+- Threshold for how soon the unhealthy state of applications can be tolerated
+    * Global Configuration
+    * Custom Configuration for specific resource types
+- Threshold for how soon a cluster can be rescheduled after being evicted
+    * Global Configuration
+    * Custom Configuration for specific clusters
+
+#### PropagationPolicy API change
+
+```
+type PropagationSpec struct {
+    ...
+    // ReschedulingTriggers represents the rescheduling config for unhealthy application state.
+    // +optional
+    ReschedulingTriggers []ReschedulingTrigger `json:"reschedulingTriggers,omitempty"`
```

Comment:
I am not sure about the default behavior which users do not set rescheduling config.
It can be `never reschedule` or `reschedule all workloads tolerating a default period of time`.
/cc @RainbowMango for opinions.